### PR TITLE
Test domain calculated properties

### PR DESCRIPTION
**Pull request by orangejenny**
Would have caught the issues fixed by https://github.com/dimagi/commcare-hq/pull/13915 and https://github.com/dimagi/commcare-hq/pull/13940

@czue / @dannyroberts

### CODE REVIEW

File: corehq/apps/domain/tests/test_domain_calculated_properties.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from __future__ import print_function, unicode_literals
+
+from django.test import TestCase
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.domain.calculations import _all_domain_stats, _calced_props
+
+class DomainCalculatedPropertiesTest(TestCase):
```
Comment (dimagimon) [inline corehq/apps/domain/tests/test_domain_calculated_properties.py]: E302 expected 2 blank lines, found 1

File: corehq/apps/domain/calculations.py [hunk 2]
```diff
@@ -332,6 +332,51 @@ def _all_domain_stats():
     }
 
 
+def _calced_props(dom, id, all_stats):
```
Comment (orangejenny) [inline corehq/apps/domain/calculations.py]: Underscore prefix seems weird, since this is imported elsewhere, but it's patterned after `_all_domain_stats`.

Comment (czue) [inline corehq/apps/domain/calculations.py]: agree it's weird. we could also change both to non-underscores

**Review by czue — APPROVED** [review]
this is great, thanks for adding

File: corehq/apps/reports/tasks.py [hunk 3]
```diff
@@ -169,55 +170,14 @@ def rebuild_export_async(config, schema):
...
-                "cp_n_sms_out_30_d": int(CALC_FNS["sms_out_in_last"](dom, 30)),
-                "cp_n_sms_out_60_d": int(CALC_FNS["sms_out_in_last"](dom, 60)),
-                "cp_n_sms_out_90_d": int(CALC_FNS["sms_out_in_last"](dom, 90)),
-                "cp_n_j2me_30_d": int(CALC_FNS["j2me_forms_in_last"](dom, 30)),
-                "cp_n_j2me_60_d": int(CALC_FNS["j2me_forms_in_last"](dom, 60)),
-                "cp_n_j2me_90_d": int(CALC_FNS["j2me_forms_in_last"](dom, 90)),
-                "cp_j2me_90_d_bool": int(CALC_FNS["j2me_forms_in_last_bool"](dom, 90)),
-                "cp_300th_form": CALC_FNS["300th_form_submission"](dom)
-            }
+            calced_props = calced_props(dom, r["_id"], all_stats)
```
Comment (dimagimon) [inline corehq/apps/reports/tasks.py]: F823 local variable 'calced_props' (defined in enclosing scope on line 34) referenced before assignment

**Comment by orangejenny** [comment]
Note to self: need to fake FormES, SMSES, etc.

https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/es/fake/README.md

**Comment by czue** [comment]
the other (easier?) option would be to just create the indices in the `setUp` function, doing something like [this](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/data_analytics/tests/test_esaccessors.py#L20).

**Comment by orangejenny** [comment]
@czue thank you, didn't know about that, will check it out

**Comment by orangejenny** [comment]
@czue Fixed tests (that was much easier than I'd expected it to be, so thanks), good to go once tests pass on travis.

**Review by czue — APPROVED** [review]
thanks!

**Pull request merged**